### ISSUE TITLE
Build fails with USE_SYSTEM_CSL=1

### ISSUE THREAD

**Issue by antonio-rojas** [JuliaLang/julia#47987]
Since commit 96592e21e2b186bfba2044dac84a15d0c31ee662 julia build fails with USE_SYSTEM_CSL=1 with error
```
ERROR: Unable to load dependent library /build/julia/src/julia-1.8.4/usr/bin/../lib/libstdc++.so.6
Message:/build/julia/src/julia-1.8.4/usr/bin/../lib/libstdc++.so.6: cannot open shared object file: No such file or directory
```
The `libstdc++.so.6` symlink is in `usr/lib/julia`, but the build system expects to find it in `usr/lib`

**Comment by t-bltg** [comment]
I've hit this too, following https://github.com/JuliaLang/julia/pull/46976.

**Comment by xgqt** [comment]
I've hit this too while packaging Julia version `1.8.4` for Gentoo,
I was thinking about applying the following workaround:

```bash
# Julia's build system still fails to copy/link GCC's libs.
mkdir -p "${S}"/usr/lib || die
local gcc_prefix
if use amd64 ; then
    gcc_prefix=x86_64-pc-linux-gnu
elif use x86 ; then
    gcc_prefix=x86-pc-linux-gnu
else
    die "Unsupported architecture"
fi
ln -s "${EROOT}"/usr/lib/gcc/${gcc_prefix}/*/libstdc++.so.6   \
    "${S}"/usr/lib/ || die	
```

But that is way to hacky and error prone.

**Comment by xgqt** [comment]
> I've hit this too while packaging Julia version `1.8.4` for Gentoo, I was thinking about applying the following workaround:
> 
> ```shell
> # Julia's build system still fails to copy/link GCC's libs.
> mkdir -p "${S}"/usr/lib || die
> local gcc_prefix
> if use amd64 ; then
>     gcc_prefix=x86_64-pc-linux-gnu
> elif use x86 ; then
>     gcc_prefix=x86-pc-linux-gnu
> else
>     die "Unsupported architecture"
> fi
> ln -s "${EROOT}"/usr/lib/gcc/${gcc_prefix}/*/libstdc++.so.6   \
>     "${S}"/usr/lib/ || die	
> ```
> 
> But that is way to hacky and error prone.

Here is the full log: [build.log](https://github.com/JuliaLang/julia/files/10369569/build.log)

**Comment by xgqt** [comment]
> I've hit this too while packaging Julia version `1.8.4` for Gentoo, I was thinking about applying the following workaround:
> 
> ```shell
> # Julia's build system still fails to copy/link GCC's libs.
> mkdir -p "${S}"/usr/lib || die
> local gcc_prefix
> if use amd64 ; then
>     gcc_prefix=x86_64-pc-linux-gnu
> elif use x86 ; then
>     gcc_prefix=x86-pc-linux-gnu
> else
>     die "Unsupported architecture"
> fi
> ln -s "${EROOT}"/usr/lib/gcc/${gcc_prefix}/*/libstdc++.so.6   \
>     "${S}"/usr/lib/ || die	
> ```
> 
> But that is way to hacky and error prone.

HMMM...
https://github.com/archlinux/svntogit-community/blob/packages/julia/trunk/PKGBUILD#L97

**Comment by fxcoudert** [comment]
We are hitting this bug in Homebrew as well: https://github.com/Homebrew/homebrew-core/pull/120170
This is currently preventing us from updating beyond 1.8.3

**Comment by staticfloat** [comment]
Can everyone affected by this issue please try out #48342 and see if that fixes anything for them?

**Comment by antonio-rojas** [comment]
> Can everyone affected by this issue please try out #48342 and see if that fixes anything for them?

Works for me

**Comment by t-bltg** [comment]
> Can everyone affected by this issue please try out https://github.com/JuliaLang/julia/pull/48342 and see if that fixes anything for them?

Thanks for fixing. Re-building `julia` from scratch with this PR applied fixes the issue.

**Comment by fxcoudert** [comment]
This fixes the build on Linux for Homebrew.

**Comment by xgqt** [comment]
For me v1.8.5 patched with https://github.com/JuliaLang/julia/pull/48342
passes compile step but fails to install.

```bash
[ ... SNIP ... ]
cp: cannot stat '/var/tmp/portage/dev-lang/julia-1.8.5/work/julia-1.8.5/usr/lib/julia/libbtf.so*': No such file or directory
chmod: cannot access '/var/tmp/portage/dev-lang/julia-1.8.5/image/usr/lib64/julia/libbtf.so*': No such file or directory
[ ... SNIP ... ]
MSYS2_ARG_CONV_EXCL='*' /var/tmp/portage/dev-lang/julia-1.8.5/work/julia-1.8.5/usr/tools/stringreplace $(strings -t x - '/var/tmp/portage/dev-lang/julia-1.8.5/image""/usr"/lib64"/libjulia.so.1.8' | grep "julia/libgcc_s.so.1:julia/libopenlibm.so:@julia/libstdc++.so.6:@libjulia-internal.so.1:@libjulia-codegen.so.1:$" | awk '{print $1;}') "julia/libgcc_s.so.1:julia/libopenlibm.so:@julia/libstdc++.so.6:@julia/libjulia-internal.so.1:@julia/libjulia-codegen.so.1:" 255 "/var/tmp/portage/dev-lang/julia-1.8.5/image""/usr"/lib64"/libjulia.so.1.8"
strings: '/var/tmp/portage/dev-lang/julia-1.8.5/image""/usr"/lib64"/libjulia.so.1.8': No such file
Usage:
  /var/tmp/portage/dev-lang/julia-1.8.5/work/julia-1.8.5/usr/tools/stringreplace <hex offset> <string to write> <maxlen> <file>
make: *** [Makefile:279: install] Error 255
```

Build log:
[build.log.txt](https://github.com/JuliaLang/julia/files/10461324/build.log.txt)

**Comment by staticfloat** [comment]
It looks to me like the critical failure in your log is:

```
strings: '/var/tmp/portage/dev-lang/julia-1.8.5/image""/usr"/lib64"/libjulia.so.1.8': No such file
```

It looks to me like you've got some strange quoting issues going on, but that's probably not the issue.  The issue is probably that it's using `lib64` instead of `lib`.  Can you share how you're overriding the lib directory name?

**Comment by vtjnash** [comment]
it is unlikely you want a relative path starting with the `"` directory

**Comment by xgqt** [comment]
> it is unlikely you want a relative path starting with the `"` directory

True, I do not want a path starting with `"`.
Makefile probably uses the variables in a wrong way.

**Comment by staticfloat** [comment]
Yeah, you should drop the quotes from your makefile overrides; Make is going to insert those, and you'll get those strange quoting issues everywhere.  This isn't a problem with our makefile, it's just how Make itself works:

```
$ cat Makefile 
FOO := "foo"

run:
        echo "$(FOO)/bar"
$ make
echo ""foo"/bar"
foo/bar
```

Note how `$(FOO)` gets expanded to `"foo"`, not `foo` as you might expect.  The shell then parses that as an empty string `""`, concatenated with `foo`, concatenated with the string `"/bar"`, which it prints out just fine, but with more complex commands you can imagine this might result in unintended quoting issues.

**Comment by xgqt** [comment]
After fixing custom Make config the install process passes.

**Comment by xgqt** [comment]
Another issue related to setting libdir it seems: 
```shell
ERROR: could not load library "/usr/bin/../lib/julia/sys.so"
/usr/bin/../lib/julia/sys.so: cannot open shared object file: No such file or directory
```

**Comment by xgqt** [comment]
> Another issue related to setting libdir it seems:
> 
> ```shell
> ERROR: could not load library "/usr/bin/../lib/julia/sys.so"
> /usr/bin/../lib/julia/sys.so: cannot open shared object file: No such file or directory
> ```

and when /usr/lib/julia is linked to /usr/lib64/julia, then Julia will fails to precompile many packages (because it fails to find libraries).

**Comment by giordano** [comment]
Fixed by #48342.
